[Swift 6.0] Windows arm64 nightly bot failing to build

https://ci-external.swift.org/job/swift-6.0-windows-toolchain-arm64/193/console

```
[2024-08-19 18:50:45] Building 'C:\Users\swift-ci\jenkins\workspace\swift-6.0-windows-toolchain-arm64\swift-foundation\Sources\FoundationMacros' to 'T:\2' for arch 'x86_64'...
-- The Swift compiler identification is Apple 6.0
<unknown>:0: warning: using (deprecated) legacy driver, Swift installation does not contain swift-driver at: 'T:\1\bin\swift-driver-new.exe'
<unknown>:0: warning: no such SDK: 'T:/arm64/Windows.platform/Developer/SDKs/Windows.sdk'
-- Check for working Swift compiler: T:/1/bin/swiftc.exe
-- Check for working Swift compiler: T:/1/bin/swiftc.exe - broken
CMake Error at C:/Program Files/CMake/share/cmake-3.29/Modules/CMakeTestSwiftCompiler.cmake:40 (message):
  The Swift compiler

    "T:/1/bin/swiftc.exe"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: 'T:/2/CMakeFiles/CMakeScratch/TryCompile-oeqjsy'
    
    Run Build Command(s): C:/PROGRA~1/MICROS~2/2022/COMMUN~1/Common7/IDE/COMMON~1/MICROS~1/CMake/Ninja/ninja.exe -v cmTC_f46f1
    [1/2][ 50%][0.087s]T:\1\bin\swiftc.exe -target x86_64-unknown-windows-msvc -j 36 -num-threads 36 -c  -module-name cmTC_f46f1 -sdk T:/arm64/Windows.platform/Developer/SDKs/Windows.sdk -gnone -Xlinker /INCREMENTAL:NO -Xlinker /OPT:REF -Xlinker /OPT:ICF  -incremental -output-file-map CMakeFiles\cmTC_f46f1.dir\\output-file-map.json  T:\2\CMakeFiles\CMakeScratch\TryCompile-oeqjsy\main.swift
    FAILED: CMakeFiles/cmTC_f46f1.dir/main.swift.obj 
    T:\1\bin\swiftc.exe -target x86_64-unknown-windows-msvc -j 36 -num-threads 36 -c  -module-name cmTC_f46f1 -sdk T:/arm64/Windows.platform/Developer/SDKs/Windows.sdk -gnone -Xlinker /INCREMENTAL:NO -Xlinker /OPT:REF -Xlinker /OPT:ICF  -incremental -output-file-map CMakeFiles\cmTC_f46f1.dir\\output-file-map.json  T:\2\CMakeFiles\CMakeScratch\TryCompile-oeqjsy\main.swift
    <unknown>:0: warning: using (deprecated) legacy driver, Swift installation does not contain swift-driver at: 'T:\1\bin\swift-driver-new.exe'
    <unknown>:0: warning: option '-incremental' is only supported in swift-driver

    <unknown>:0: warning: no such SDK: 'T:/arm64/Windows.platform/Developer/SDKs/Windows.sdk'

    <unknown>:0: error: unable to load standard library for target 'x86_64-unknown-windows-msvc'

    ninja: build stopped: subcommand failed.
    
    

  

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:26 (project)


-- Configuring incomplete, errors occurred!
Error: Error: cmake.exe exited with code 1.
Invocation:
  cmake.exe -B T:\2 -S C:\Users\swift-ci\jenkins\workspace\swift-6.0-windows-toolchain-arm64\swift-foundation\Sources\FoundationMacros -G Ninja -D CMAKE_BUILD_TYPE=Release -D CMAKE_MT=mt -D CMAKE_Swift_COMPILER=T:/1/bin/swiftc.exe -D CMAKE_Swift_COMPILER_TARGET=x86_64-unknown-windows-msvc -D CMAKE_Swift_FLAGS=-sdk T:/arm64/Windows.platform/Developer/SDKs/Windows.sdk -gnone -Xlinker /INCREMENTAL:NO -Xlinker /OPT:REF -Xlinker /OPT:ICF -D CMAKE_Swift_FLAGS_RELEASE=-O -D CMAKE_Swift_FLAGS_RELWITHDEBINFO=-O -D SwiftSyntax_DIR=T:/5/cmake/modules
```

--- a/utils/build.ps1
+++ b/utils/build.ps1
@@ -1752,7 +1752,7 @@ function Build-FoundationMacros() {
 
   $SwiftSDK = $null
   if ($Build) {
-    $SwiftSDK = $HostArch.SDKInstallRoot
+    $SwiftSDK = $BuildArch.SDKInstallRoot
   }
 
   $Targets = if ($Build) {
@@ -1766,6 +1766,12 @@ function Build-FoundationMacros() {
     $InstallDir = "$($Arch.ToolchainInstallRoot)\usr"
   }
 
+  $SwiftSyntaxCMakeModules = if ($Build -and $HostArch -ne $BuildArch) {
+    Get-BuildProjectCMakeModules Compilers
+  } else {
+    Get-HostProjectCMakeModules Compilers
+  }
+
   Build-CMakeProject `
     -Src $SourceCache\swift-foundation\Sources\FoundationMacros `
     -Bin $FoundationMacrosBinaryCache `
@@ -1776,7 +1782,7 @@ function Build-FoundationMacros() {
     -SwiftSDK:$SwiftSDK `
     -BuildTargets $Targets `
     -Defines @{
-      SwiftSyntax_DIR = (Get-HostProjectCMakeModules Compilers);
+      SwiftSyntax_DIR = $SwiftSyntaxCMakeModules;
     }
 }
 

